### ISSUE TITLE
git state exception on branch change

### ISSUE THREAD

**Issue by ekle** [saltstack/salt#29783]
minion version is 2015.8.3+ds-1 on debian 8
```yaml
test-git:
  git.latest:
    - name: ssh://git@host/abc.git
    - rev: master
    - force_reset: true
    - target: '{{ project_directory }}'
    - identity: '/user/.ssh/id_rsa'
    - force_checkout: True
    - force_fetch: True
    - force_reset: True
    - require:
      - pkg: git
```
```
==> devbox:      Comment: An exception occurred in this state: Traceback (most recent call last):
==> devbox:                 File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1594, in call  
==> devbox:                   **cdata['kwargs'])                                                       
==> devbox:                 File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1491, in wrapper                                                                                                     
==> devbox:                   return f(*args, **kwargs)                                                
==> devbox:                 File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 1195, in latest                                                                                                  
==> devbox:                   return _uptodate(ret, target, comments)                                  
==> devbox:                 File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 107, in _uptodate                                                                                                
==> devbox:                   ret['comment'] += '\n\nChanges made: ' + comments                        
==> devbox:               TypeError: cannot concatenate 'str' and 'list' objects      
```

**Comment by vitaliyf** [comment]
Looks like this was fixed in https://github.com/saltstack/salt/commit/3c4185f806955dc263b2c886408e6d77c25eeccb

**Comment by rallytime** [comment]
@ekle Are you in a position to give the fix above a try? That fix should be coming out in the 2015.8.4 release shortly.

**Comment by basepi** [comment]
I'm going to close this. Keep us posted if it's not fixed.
